hamster_gtk/preferences/widgets/combo_file_chooser.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+
+# This file is part of 'hamster-gtk'.
+#
+# 'hamster-gtk' is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# 'hamster-gtk' is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with 'hamster-gtk'.  If not, see <http://www.gnu.org/licenses/>.
+
+"""This module provides file chooser widget."""
+
+# [FIXME]
+# Adding 'unicode_literals' raises encoding issues. This is a major sign we
+# have a unicode issue!
+from __future__ import absolute_import
+
+from gettext import gettext as _
+
+from gi.repository import Gtk
+from six import text_type
+
+from hamster_gtk.helpers import get_parent_window
+
+# [FIXME]
+# Remove once hamster-lib has been patched
+from hamster_gtk.helpers import _u
+
+from .config_widget import ConfigWidget
+
+
+class ComboFileChooser(Gtk.Grid, ConfigWidget):
+    """A file chooser that also has an entry for changing the path."""
+
+    # Required else you would need to specify the full module name in ui file
+    __gtype_name__ = 'ComboFileChooser'
+
+    def __init__(self):
+        """Initialize widget."""
+        super(Gtk.Grid, self).__init__()
+
+        self._entry = Gtk.Entry()
+        self._entry.set_hexpand(True)
+
+        self._button = Gtk.Button(_("Choose"))
```

Comment:
No, I do not use mnemonics here as it would probably be confusing. It would be the same for all the buttons in Preferences and would cycle between them.

People can access the entry using mnemonics and then <kbd>tab</kbd> to use the button. Maybe the button could be focused when `_on_mnemonic_activate` is called, though accessing the entry would then require <kbd>shift</kbd>-<kbd>tab</kbd>.